src/Switch/Switch.js

```diff
@@ -57,17 +58,36 @@ export const styleSheet = createStyleSheet('Switch', (theme) => {
 });
 
 export default function Switch(props, context) {
-  const { className, checkedClassName, ...other } = props;
+  const { className, checkedClassName, label, labelClassName, labelReverse, ...other } = props;
   const classes = context.styleManager.render(styleSheet);
+
+  const switchProps = {
+    className: classes.default,
+    checkedClassName: classNames(classes.checked, checkedClassName),
+    icon: <div className={classes.icon} />,
+    checkedIcon: <div className={classes.iconChecked} />,
+    type: 'checkbox',
+    ...other,
+  };
+
+  if (label) {
+    switchProps['aria-label'] = label;
+    return (
+      <SelectionLabel label={label} className={labelClassName} labelReverse={labelReverse}>
+        <div className={classNames(classes.root, className)}>
+          <SwitchBase
+            {...switchProps}
+          />
```

Comment:
You had me change the other selection control to this from what you are suggesting; I'll switch them back.